test/ApplicationInsightsJavaScriptSnippetTest/Validator.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace ApplicationInsightsJavaScriptSnippetTest
+{
+    public class Validator
+    {
+        private HttpClient _httpClient;
+
+        private HttpClientHandler _httpClientHandler;
+
+        private readonly ILogger _logger;
+
+        private readonly DeploymentResult _deploymentResult;
+
+        public Validator(
+            HttpClient httpClient,
+            HttpClientHandler httpClientHandler,
+            ILogger logger,
+            DeploymentResult deploymentResult)
+        {
+            _httpClient = httpClient;
+            _httpClientHandler = httpClientHandler;
+            _logger = logger;
+            _deploymentResult = deploymentResult;
+        }
+
+        public async Task VerifyScriptCheckPage(HttpResponseMessage response)
+        {
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                _logger.LogInformation("Home page content : {0}", responseContent);
+            }
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            ValidateLayoutPage(responseContent);
+        }
+
+        private void ValidateLayoutPage(string responseContent)
+        {
+            var path = Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "test\\ApplicationInsightsJavaScriptSnippetTest\\Rendered.html"));
```

Comment:
1. Can use `@"foo\bar\baz` to avoid gnarly escaping
2. But don't even do that. Instead of reading the file from disk, embed it as a resource, and read the resource. This avoids weird path issues.